x/superfluid/keeper/slash.go

```diff
@@ -59,11 +63,72 @@ func (k Keeper) SlashLockupsForValidatorSlash(ctx sdk.Context, valAddr sdk.ValAd
+
+	// If the slashCoins contains a cl denom, we need to update the underlying cl position to reflect the slash.
 	_ = osmoutils.ApplyFuncIfNoError(ctx, func(cacheCtx sdk.Context) error {
-		// These tokens get moved to the community pool.
-		_, err := k.lk.SlashTokensFromLockByID(cacheCtx, lock.ID, slashCoins)
-		return err
+		if strings.HasPrefix(lock.Coins[0].Denom, cltypes.ClTokenPrefix) {
+			// Run prepare logic to get the underlying coins to slash.
+			// We get the pool address here since the underlying coins will be sent directly from the pool to the community pool instead of the lock module account.
+			// Additionally, we update the cl position's state entry to reflect the slash in the position's liquidity.
+			poolAddress, underlyingCoinsToSlash, err := k.prepareConcentratedLockForSlash(cacheCtx, lock, slashAmt)
+			if err != nil {
+				return err
+			}
+			// Run the normal slashing logic, but instead of sending gamm shares to the community pool, we send the underlying coins
+			// the cl shares represent to the community pool and burn the cl shares from the lockup module account as well as the lock itself
+			_, err = k.lk.SlashTokensFromLockByIDSendUnderlyingAndBurn(cacheCtx, lock.ID, lockSharesToSlash, underlyingCoinsToSlash, poolAddress)
+			return err
+		} else {
+			// These tokens get moved to the community pool.
+			_, err := k.lk.SlashTokensFromLockByID(cacheCtx, lock.ID, lockSharesToSlash)
+			return err
+		}
 	})
 }
+
+// prepareConcentratedLockForSlash is a helper function that runs pre-slash logic for concentrated lockups. This function:
+// 1. Figures out the underlying assets from the liquidity being slashed and creates a coin object this represents
+// 2. Sets the cl position's liquidity state entry to reflect the slash
+// 3. Returns the pool address that will sends the underlying coins as well as the underlying coins to slash
+func (k Keeper) prepareConcentratedLockForSlash(ctx sdk.Context, lock *lockuptypes.PeriodLock, slashAmt sdk.Dec) (sdk.AccAddress, sdk.Coins, error) {
+	// Get the position ID from the lock denom
+	positionID, err := cltypes.GetPositionIdFromShareDenom(lock.Coins[0].Denom)
+	if err != nil {
+		return sdk.AccAddress{}, sdk.Coins{}, err
+	}
+
+	// Figure out the underlying assets from the liquidity slash
+	position, err := k.clk.GetPosition(ctx, positionID)
+	if err != nil {
+		return sdk.AccAddress{}, sdk.Coins{}, err
+	}
+	slashAmtNeg := slashAmt.Neg()
+
+	// If slashAmt is not negative, return an error
+	if slashAmtNeg.IsPositive() {
+		return sdk.AccAddress{}, sdk.Coins{}, fmt.Errorf("slash amount must be negative, got %s", slashAmt)
+	}
+
+	position.Liquidity = slashAmt
+	concentratedPool, err := k.clk.GetPoolFromPoolIdAndConvertToConcentrated(ctx, position.PoolId)
+	if err != nil {
+		return sdk.AccAddress{}, sdk.Coins{}, err
+	}
+	asset0, asset1, err := cl.CalculateUnderlyingAssetsFromPosition(ctx, position, concentratedPool)
```

Comment:
I think we might want to change `CalculateUnderlyingAssetsFromPosition` to take multiple parameters corresponding to liquidity ticks etc

The approach of creating a temporary position with slash amount isn't ideal IMO

Reply:
https://github.com/osmosis-labs/osmosis/pull/4884/commits/b16e16eb2c6e74b25cdc29592353acfa9bc288ad